### PR TITLE
IDE-3204 bundle buildship 2.0.1 and change api and fix tests

### PR DESCRIPTION
**Pull request by AndyWu2015**
I am shipping snapshot version of buildship and buildship will release 2.1 next Wensday , then we can bundle that.

### CODE REVIEW

**Comment by gamerson** [comment]
hey Andy, do you think its stable to move to buildship 2.0.1 before we go to GA?  I guess it is, what do you think @jtydhr88 

But andy we have a merge error now, can you rebase and resend?

File: build/parent/pom.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@
         <eclipse-xml-search-site>https://dl.bintray.com/gamerson/eclipse-wtp-xml-search/</eclipse-xml-search-site>
         <tern-site>http://oss.opensagres.fr/tern.repository/1.2.1/</tern-site>
         <bndtools-site>https://bndtools.ci.cloudbees.com/job/bndtools.master/lastSuccessfulBuild/artifact/build/generated/p2/</bndtools-site>
-        <gradle-site>http://download.eclipse.org/buildship/updates/e46/releases/1.0/1.0.21.v20161010-1640</gradle-site>
+        <gradle-site>http://download.eclipse.org/buildship/updates/e46/snapshots/2.x/</gradle-site>
```
Comment (gamerson) [inline build/parent/pom.xml]: Lets make it a specific version, not, 2.x, and lets do it from the 'releases' not 'snapshots'

File: tools/plugins/com.liferay.ide.gradle.core/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -13,12 +13,12 @@ Require-Bundle: org.eclipse.core.runtime,
  org.eclipse.debug.core,
  org.eclipse.jface,
  biz.aQute.bndlib;bundle-version="3.3.0",
- org.eclipse.buildship.core;bundle-version="1.0.21",
- com.gradleware.tooling.model;bundle-version="0.16.0",
+ org.eclipse.buildship.core,
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/META-INF/MANIFEST.MF]: Lets put a verison here, so we make sure we are using the 2.0.xx versions of the core plugin.

File: tools/plugins/com.liferay.ide.gradle.core/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -13,12 +13,12 @@ Require-Bundle: org.eclipse.core.runtime,
...
+ com.gradleware.tooling.model,
  com.liferay.ide.project.core,
  com.liferay.blade.api;bundle-version="2.0.0",
- com.gradleware.tooling.client;bundle-version="0.16.0",
- com.gradleware.tooling.utils;bundle-version="0.16.0",
+ com.gradleware.tooling.client,
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/META-INF/MANIFEST.MF]: Also lets add versions here, so we can make sure we are using the right versions.

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java [hunk 4]
```diff
@@ -111,9 +96,9 @@ public static GradleCore getDefault()
         return retval;
     }
 
-    public static <T> T getToolingModel( GradleCore gradleCore, Class<T> modelClass, IProject gradleProject )
+    public static <T> T getToolingModel( Class<T> modelClass, IProject gradleProject )
     {
-        return getToolingModel( gradleCore, modelClass, gradleProject.getLocation().toFile() );
+        return getToolingModel( modelClass, gradleProject.getLocation().toFile() );
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java]: Is there any API that we can keep from having to build our own tooling model?

Comment (AndyWu2015) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java]: not sure yet,  I will do some research

File: build/parent/pom.xml (hunk 1 above)
Comment (AndyWu2015) [inline build/parent/pom.xml]: build will release 2.1 two days later , then we can bundle that release version.
see 
release date

File: tools/plugins/com.liferay.ide.gradle.core/META-INF/MANIFEST.MF (hunk 3 above)
Comment (AndyWu2015) [inline tools/plugins/com.liferay.ide.gradle.core/META-INF/MANIFEST.MF]: got it

**Pull request closed without merging**